src/alphafold_impact/pipelines/data_collection_oa/pipeline.py

```diff
@@ -0,0 +1,98 @@
+"""
+Pipeline for collecting incoming (cites) or outgoing (references) citations for a given work.
+
+The main pipeline is a template pipeline that is used to create two pipelines, using the 
+`modular_pipeline` function. The two pipelines are `incoming_pipeline` and
+`outgoing_pipeline`, which are used to collect incoming and outgoing citations, respectively.
+
+Note that the output folders are also dynamically named based on the direction of the
+citations.
+
+To run this pipeline, use the following command:
+    
+        $ kedro run --pipeline data_collection_oa
+
+To run this pipeline with a specific work ID, use the following command:
+
+        $ kedro run --pipeline data_collection_oa --params get.work_id=<work_id> 
+        (ie. --params get.work_id=W2741809807)
+
+To run this pipeline with a specific direction, use the following command:
+
+        $ kedro run --pipeline data_collection_oa --tags <direction> (ie. --tags incoming)
+
+The pipeline also includes a `downstream_impact_pipeline` that is used to collect the
+incoming citations for the papers that cite the given work ID. This pipeline is used to
+calculate the downstream impact of a given work by collecting data for all subsequent
+work that cites the work_id.
+
+To run this pipeline, use the following command:
+    
+            $ kedro run --pipeline data_collection_oa --tags downstream_impact
+
+The pipeline will save the data to the following location:
+            s3://alphafold-impact/data/02_intermediate/openalex/works/cited_by/downstream/<work_id>
+using a partitioned dataset.
+"""
+
+from kedro.pipeline import Pipeline, node, pipeline
+from kedro.pipeline.modular_pipeline import pipeline as mpl
+
+from .nodes import collect_citation_papers, load_work_ids
+
+template_pipeline = pipeline(
+    [
+        node(
+            func=collect_citation_papers,
+            inputs={
+                "mailto": "params:api.mailto",
+                "perpage": "params:api.perpage",
+                "work_ids": "params:get.work_id",
+                "direction": "params:direction.template",
+            },
+            outputs="template",
+        )
+    ]
+)
+
+incoming_pipeline = mpl(
+    pipe=template_pipeline,
+    parameters={"params:direction.template": "params:direction.incoming"},
+    outputs={"template": "works_outgoing_citations"},
+    tags="incoming",
+)
+
+outgoing_pipeline = mpl(
+    pipe=template_pipeline,
+    parameters={"params:direction.template": "params:direction.outgoing"},
+    outputs={"template": "works_incoming_citations"},
+    tags="outgoing",
+)
```

Comment:
The modular pipelines look like they could be useful 🙂

Reply:
Indeed. It wasn't necessary here (no one asked me to have a pipeline for `cites`, but I needed to have a working example to see what it did. Well, now we do :)